src/annotationForm/AnnotationDrawing.js

```diff
@@ -43,22 +60,100 @@ function AnnotationDrawing({ drawingState, setDrawingState, height, width, ...pr
       currentShape: shape,
       shapes: [...drawingState.shapes, shape],
     });
-  }, [props.imageEvent]);
+  }, [drawingState, imageEvent, props.imageEvent, setDrawingState]);
```

Comment:
Why did you add these new dependancy in useEffect ?
UseEffect can be removed I thing (need to check)